src/components/Checkbox/Checkbox.tsx

```diff
@@ -0,0 +1,72 @@
+import React, { useState } from "react";
+import { nanoid } from "nanoid";
+import Field from "../internal/Field";
+import InternalCheckbox from "../internal/InternalCheckbox";
+import { defaultCheckboxProps } from "./defaultCheckboxProps";
+import { InternalCheckboxProps } from "./types";
+import { useMergedProps } from "../../hooks/useMergedProps";
+
+function Checkbox(props: InternalCheckboxProps) {
+  const mergedProps = useMergedProps(props, defaultCheckboxProps);
+  const {
+    label,
+    value,
+    hideLabel,
+    helpText,
+    disabled,
+    optional,
+    onChange,
+    onFocus,
+    onBlur,
+    onMouseDown,
+    children,
+    testId,
+    error = [],
+    __internal__keyboardFocus,
+  } = mergedProps;
+  const [labelId] = useState(() => `radio-group-label-${nanoid()}`);
+  const [inputId] = useState(() => `checkbox-${nanoid()}`);
+  const [auxId] = useState(() => `checkbox-aux-${nanoid()}`);
+
+  const onCheckboxChange: React.ChangeEventHandler<HTMLInputElement> = (
+    event
+  ) => {
+    onChange?.(event.target.checked);
```

Comment:
For readabilities sake?

```suggestion
    if(onChange) onChange(event.target.checked);
```